docs/index.md

```diff
@@ -367,6 +367,27 @@ Adding errors to action object is important, because each time a handled excepti
 Validation exception will have the same backtrace as original error.
 Prefer this way over custom exception handling in private methods.
 
+### Dependency Injection
+
+Granite uses action initializer for attributes.
+However, we can still manually inject dependencies.
```

Comment:
I don't think we should be saying that this is how you should be doing dependency injection. I'm pretty sure that something like:
```ruby
Action.attribute :my_dep, Object
Action.new(my_dep: Bar.new)
```
Would also work. While I do see value in documenting ability to use custom initializers, something like: "here's how you can use custom parameters in cases where you don't want your data to end up in `action.attributes` (for example when using dependency injection)".

Reply:
@andriusch You're mixing up attributes with dependencies. Which is a big problem with current Granite implementation.

Reply:
Sorry I think I didn't explain myself properly. I didn't mean that using `attributes` for DI is the correct approach, merely that it's a possible approach. My main concern here is that you're documenting only in regards to DI, but custom initializers could be used for other things as well, not only DI. So I'd prefer if it was worded something like: "You can still use custom initializers for Granite actions when you don't want to use attributes, for example when using dependency injection:". That way we're still documenting the preferred way to use DI, but also make it clear that this is not limited to DI.

Reply:
Fixed in https://github.com/toptal/granite/pull/64/commits/57011bdcd89d18e1df00c9413582937c4cddc941